packages/browser/src/plugins/middleware/index.ts

```diff
@@ -39,26 +41,24 @@ export async function applyDestinationMiddleware(
     let nextCalled = false
     let returnedEvent: SegmentEvent | null = null
 
-    await asPromise(
-      fn({
-        payload: toFacade(event, {
-          clone: true,
-          traverse: false,
-        }),
-        integration: destination,
-        next(evt) {
-          nextCalled = true
-
-          if (evt === null) {
-            returnedEvent = null
-          }
-
-          if (evt) {
-            returnedEvent = evt.obj
-          }
-        },
-      })
-    )
+    await fn({
```

Comment:
I assume that the reason you need to wrap `fn` in an asPromise wrapper was introduced is because `MiddlewareFunction` and `SourceMiddleware` could actually return a Promise<void> -- but maybe they didn't realize  you could have a return type of void | Promise<void>. And, they might not have understood that there's no requirement to "normalize" all values to a promise in order to await -- that you can just as easily await a regular value as a promise.